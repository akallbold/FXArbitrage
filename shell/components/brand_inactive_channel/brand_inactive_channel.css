.container {
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: calc(100vh - 64px);
}

.innerContainer {
  width: 300px;
  text-align: center;
}

.description {
  width: 300px;
  margin-bottom: 20px;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.button {
  display: inline-block;
  width: 220px;
}

/* Non Admin styles */
:global(.isAnalyst) .button {
	opacity: 0.5;
	pointer-events: none;
}