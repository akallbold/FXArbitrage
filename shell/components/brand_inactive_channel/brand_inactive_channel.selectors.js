import { createSelector } from 'reselect'
import { brandChannels } from 'selectors/org_selectors'
import { curChannelSelector } from 'selectors/platform_selectors'

const brandHasChannelSelector = createSelector(
  brandChannels,
  curChannelSelector,
  (channels, curChannel) => (channels[curChannel])
)

export default brandHasChannelSelector
