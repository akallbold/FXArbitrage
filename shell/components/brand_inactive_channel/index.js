export CheckBrandChannelActive from './brand_inactive_channel'
