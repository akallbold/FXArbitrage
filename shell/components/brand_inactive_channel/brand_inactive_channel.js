import React from 'react'
import PropTypes from 'prop-types'
import { startCase } from 'lodash'
import { connect } from 'react-redux'
import { Button, Modal, SpinnerModalContent } from 'playa-styleguide'
import { curChannelSelector } from 'selectors/platform_selectors'
import { orgsLoadingStatusSelector } from 'selectors/org_selectors'
import brandHasChannelSelector from './brand_inactive_channel.selectors'
import * as styles from './brand_inactive_channel.css'

// We need to link to the legacy configuration until the settings section is
// implemented. TODO: replace with link to new settings once implemented.
const legacySetupUrl = `${process.env.FE_APP_HOST}/settings/social_profiles`

export const BrandInactiveChannel = ({ channel }) => {
  const chanName = startCase(channel)
  return (
    <div className={styles.container}>
      <div className={styles.innerContainer}>
        <div className={styles.description}>This brand doesn&apos;t advertise on {chanName}</div>
        <Button size='large' href={legacySetupUrl} className={styles.button}>Add to {chanName}</Button>
      </div>
    </div>
  )
}

BrandInactiveChannel.propTypes = { channel: PropTypes.string }
BrandInactiveChannel.defaultProps = { channel: undefined }

export const CheckBrandChannelActive = ({ orgsLoadingStatus, brandHasChannel, channel, children }) => (
  <span>
    {/* Still loading orgs data */}
    {orgsLoadingStatus &&
      <Modal isOpen closeTimeoutMS={500}>
        <SpinnerModalContent />
      </Modal>
    }
    {/* Done loading orgs and its a valid brand channel */}
    {!orgsLoadingStatus && brandHasChannel &&
      children
    }
    {/* Done loading orgs and its an invalid brand channel */}
    {!orgsLoadingStatus && !brandHasChannel &&
      <BrandInactiveChannel channel={channel} />
    }
  </span>
)

CheckBrandChannelActive.propTypes = {
  orgsLoadingStatus: PropTypes.bool,
  brandHasChannel: PropTypes.bool,
  channel: PropTypes.string,
  children: PropTypes.node
}

CheckBrandChannelActive.defaultProps = {
  orgsLoadingStatus: false,
  brandHasChannel: false,
  channel: undefined,
  children: undefined
}

const mapStateToProps = state => ({
  orgsLoadingStatus: orgsLoadingStatusSelector(state),
  brandHasChannel: brandHasChannelSelector(state),
  channel: curChannelSelector(state)
})

export default connect(mapStateToProps)(CheckBrandChannelActive)
