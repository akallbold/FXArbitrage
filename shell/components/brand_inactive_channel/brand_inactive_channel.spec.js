import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrandInactiveChannel, CheckBrandChannelActive } from './brand_inactive_channel'

describe('<CheckBrandChannelActive />', () => {
  it('renders the children when brandHasChannel is true', () => {
    const component = shallow(
      <CheckBrandChannelActive brandHasChannel>
        <div id='child'>We are the children</div>
      </CheckBrandChannelActive>
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders the BrandInactiveChannel when brandHasChannel is false', () => {
    const component = shallow(
      <CheckBrandChannelActive channel='twitter'>
        <div id='child'>We are the children</div>
      </CheckBrandChannelActive>
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders loading modal when orgs are loading', () => {
    const component = shallow(
      <CheckBrandChannelActive orgsLoadingStatus>
        <div id='child'>We are the children</div>
      </CheckBrandChannelActive>
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('<BrandInactiveChannel />', () => {
  it('renders', () => {
    const component = shallow(<BrandInactiveChannel channel='twitter' />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
