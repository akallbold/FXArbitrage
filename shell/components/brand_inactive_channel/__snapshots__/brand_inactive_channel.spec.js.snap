// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandInactiveChannel /> renders 1`] = `
<div>
  <div>
    <div>
      This brand doesn't advertise on 
      Twitter
    </div>
    <Button
      className=""
      component={null}
      disabled={false}
      href="undefined/settings/social_profiles"
      onClick={null}
      priority="primary"
      size="large"
    >
      Add to 
      Twitter
    </Button>
  </div>
</div>
`;

exports[`<CheckBrandChannelActive /> renders loading modal when orgs are loading 1`] = `
<span>
  <Modal
    closeTimeoutMS={500}
    contentLabel="this is a modal"
    handleCloseClick={null}
    isOpen={true}
  >
    <SpinnerModalContent
      error={false}
      errorOnClose={null}
      message={null}
      success={false}
      successOnClose={null}
    />
  </Modal>
</span>
`;

exports[`<CheckBrandChannelActive /> renders the BrandInactiveChannel when brandHasChannel is false 1`] = `
<span>
  <BrandInactiveChannel
    channel="twitter"
  />
</span>
`;

exports[`<CheckBrandChannelActive /> renders the children when brandHasChannel is true 1`] = `
<span>
  <div
    id="child"
  >
    We are the children
  </div>
</span>
`;
